pkg/metadata/scheduler.go

```diff
@@ -64,8 +70,11 @@ func (c *Scheduler) AddCollector(name string, interval time.Duration) error {
 	health := health.Register("metadata-" + name)
 
 	go func() {
+		ctx := context.Context(c.context)
```

Comment:
in order to create a new context, with the cancel propagation configured, you need to use `context.WithCancel(...)`
(doc: https://golang.org/src/context/context.go?s=8226:8290#L219)

```suggestion
		ctx, stopCtxFunc := context.WithCancel(c.context)
		defer stopCtxFunc()
```